src/command-processor.js

```diff
@@ -36,8 +36,18 @@ export async function processMessage(msg) {
                 return "I have recorded your progress friend.";
             case STATS_CMD:
                 const pastWeights = await weightStats(msg.author.username);
-                const average = await averageWeight(pastWeights);
-                return `Past 7 Weights: ${pastWeights}\n7 Day Average: ${average}`;
+                const reversedWeights = pastWeights.reverse()
+                const currentAvg = averageWeight(reversedWeights.slice(-7));
+                const lastAvg = averageWeight(reversedWeights.slice(0,7));
+                const average = averageWeight(reversedWeights.slice(-7));
+                const averageDifference = (lastAvg - currentAvg).toPrecision(5);
+                let direction;
+                if (averageDifference <= 0) {
+                    direction = "Gained";
+                  } else {
+                    direction = "Lost";
+                  }
+                return `Past 7 Weights: ${pastWeights} lbs.\n7 Day Average: ${average} lbs.\nAverage difference: ${direction} ${averageDifference} lbs.`;
```

Comment:
Should we omit trend calculation if we do not have data for 14 or more days?

I also think we will need to bucket by dates - my rationale is users may miss updating weights on a few dates and we should calculate the average based on data we have for the past two weeks instead of including data points that may slide in from before - we don't have to fix this with this CR but we can work on future updates to address this.

Reply:
Added conditional statements for the 14-day average. Will look into missing dates next. Currently assumes no missing dates.